design_notes/XDN05 - xlang Binary Interface.md

```diff
@@ -194,15 +200,17 @@ of this document. We need a design note on this.
 While all information categories are optional, it is highly recommended that all objects support
 TypeName in order to be usable from dynamic languages.
 
-#### String Representation
+#### Hash Code
+
+The HashCode information category retrieves a numeric value that is used to identify an object in a
+hash based collectionThe property type for StringRepresentation is a 32 bit integer.
 
-Most mainstream languages including
-[JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString),
-[Python](https://docs.python.org/3/library/stdtypes.html#str),
-[.NET](https://docs.microsoft.com/en-us/dotnet/api/system.object.tostring?view=netframework-4.7.2),
-[Java](https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#toString()) and
-[Objective-C](https://developer.apple.com/documentation/objectivec/1418956-nsobject/1418746-description)
-have a mechanism to retrieve the string representation of an object.
+This information category should only be supported for types that wish to return a custom value for
+a language projection's equivalent of GetHashCode. Language projections may choose to fall back to
+a hash code derived from the object's reference identification (such as a C++ pointer) if HashCode
```

Comment:
If we build components in langauges other than C++, it may not be a pointer. Python objects have an id, which happens to be the object's address in memory on the CPython implementation, but that's an implementation detail.